webhooks/virtualmachine/v1alpha2/mutation/virtualmachine_mutator_intg_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2019-2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package mutation_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha2"
+	"github.com/vmware-tanzu/vm-operator/test/builder"
+)
+
+func intgTests() {
+	Describe("Invoking Mutation", intgTestsMutating)
+}
+
+type intgMutatingWebhookContext struct {
+	builder.IntegrationTestContext
+	vm *vmopv1.VirtualMachine
+}
+
+func newIntgMutatingWebhookContext() *intgMutatingWebhookContext {
+	ctx := &intgMutatingWebhookContext{
+		IntegrationTestContext: *suite.NewIntegrationTestContext(),
+	}
+
+	ctx.vm = builder.DummyVirtualMachineA2()
+	ctx.vm.Namespace = ctx.Namespace
+
+	return ctx
+}
+
+func intgTestsMutating() {
+	var (
+		ctx *intgMutatingWebhookContext
+	)
+
+	BeforeEach(func() {
+		ctx = newIntgMutatingWebhookContext()
+	})
+
+	AfterEach(func() {
+		ctx = nil
+	})
+
+	Context("SetNextRestartTime", func() {
+		When("create a VM", func() {
+			When("spec.nextRestartTime is empty", func() {
+				It("should not mutate", func() {
+					ctx.vm.Spec.NextRestartTime = ""
+					Expect(ctx.Client.Create(ctx, ctx.vm)).To(Succeed())
+					modified := &vmopv1.VirtualMachine{}
+					Expect(ctx.Client.Get(ctx, client.ObjectKeyFromObject(ctx.vm), modified)).Should(Succeed())
+					Expect(modified.Spec.NextRestartTime).Should(BeEmpty())
+				})
+			})
+			When("spec.nextRestartTime is not empty", func() {
+				It("should not mutate", func() {
+					ctx.vm.Spec.NextRestartTime = "hello"
+					Expect(ctx.Client.Create(ctx, ctx.vm)).To(Succeed())
+					modified := &vmopv1.VirtualMachine{}
+					Expect(ctx.Client.Get(ctx, client.ObjectKeyFromObject(ctx.vm), modified)).Should(Succeed())
+					Expect(modified.Spec.NextRestartTime).Should(Equal("hello"))
+				})
+			})
+		})
+
+		When("updating VirtualMachine", func() {
+			var (
+				lastRestartTime time.Time
+				modified        *vmopv1.VirtualMachine
+			)
+			BeforeEach(func() {
+				lastRestartTime = time.Now().UTC()
+				modified = &vmopv1.VirtualMachine{}
+			})
+			JustBeforeEach(func() {
+				Expect(ctx.Client.Create(ctx, ctx.vm)).To(Succeed())
+				Expect(ctx.Client.Get(ctx, client.ObjectKeyFromObject(ctx.vm), modified)).Should(Succeed())
+			})
+			When("spec.nextRestartTime is empty", func() {
```

Comment:
```suggestion
			When("existing spec.nextRestartTime is empty", func() {
```

Reply:
I'm keeping this identical to v1a1 tests